proto/proto_annotated.exs

```diff
@@ -0,0 +1,121 @@
+    # Maybe there's a special formatting that I'm not aware of for a big match?
+    << magic_string :: binary-size(4), version :: integer, num_records :: big-integer-32, rest :: binary >> = binary
+
+    handleRecords(rest, %{
+      magic_string: magic_string,
+      version: version,
+      num_records: num_records,
+      records: [],
+      debit_total: 0,
+      credit_total: 0,
+      startautopay_count: 0,
+      endautopay_count: 0,
+    })
+  end
+
+  # I feel like maybe this portion would be better as a case?
+  def handleRecords(<< 0, rest :: binary >>, acc) do
+    handleFourField("Debit", rest, acc)
+  end
+  def handleRecords(<< 1, rest :: binary >>, acc) do
+    handleFourField("Credit", rest, acc)
+  end
+  def handleRecords(<< 2, rest :: binary >>, acc) do
+    handleThreeField("StartAutopay", rest, acc)
+  end
+  def handleRecords(<< 3, rest :: binary >>, acc) do
+    handleThreeField("EndAutopay", rest, acc)
+  end
+  def handleRecords("", acc) do
+    displayResults(acc)
+  end
+
+  def handleFourField(type, binary, acc) do
+    << timestamp :: big-integer-32, user_id :: big-integer-64, amount :: float, rest :: binary >> = binary
+
+    record = %{
+      type: type,
+      timestamp: timestamp,
+      user_id: user_id,
+      amount: amount
+    }
+
+    # This felt dirty, I suppose it depends on the audience. As an alternative
+    # I could see matching on the "Debit" or "Credit" and passing an extra arg
+    # as the key...
+    #
+    # Another thing that kinda put me on edge when I was figuring out how to
+    # update a Map, what if I didn't want a default in case the lookup failed?
+    # I suppose you provide the least harmful default. There is also the
+    # possibility that my problem is one of mindset, that I'm too used to the
+    # wild-west and the restraints on this v-small system would be such that
+    # I'd never be making new entries just for the lolz on accident.
+    type_key = String.to_atom("#{String.downcase(type)}_total")
+
+    # Is there a more elegant way to alter acc? Maybe do the pipelines in the
+    # param list of handleRecords?
+    acc = acc 
+          |> Map.update(:records, [], &[ record | &1 ])
+          |> Map.update(type_key, 0, &( &1 + amount ))
+
+    handleRecords(rest, acc)
+  end
+
+  def handleThreeField(type, binary, acc) do
+    << timestamp :: big-integer-32, user_id :: big-integer-64, rest :: binary >> = binary
+
+    record = %{
+      type: type,
+      timestamp: timestamp,
+      user_id: user_id
+    }
+
+    type_key = String.to_atom("#{String.downcase(type)}_count")
+
+    acc = acc 
+          |> Map.update(:records, [], &[ record | &1 ])
+          |> Map.update(type_key, 0, &( &1 + 1 ))
+
+    handleRecords(rest, acc)
+  end
+
+  def displayResults(acc) do
+    special_user_record = Enum.find(acc.records, &( &1.user_id == 2456938384156277127 ))
+    actual_number_records = length(acc.records)
+
+    IO.puts "============ Results ============="
```

Comment:
I think with an example this small, this output works OK, but I wanted to tell you about some other options and avenues for writing: 

### [Artificery](https://hex.pm/packages/artificery): 
Namely [`Artificery.Console.Table.format/4`](https://hexdocs.pm/artificery/Artificery.Console.Table.html#format/4) for creating tables. It's nice and clean. It also is another dependency which might not be acceptable for use in a script. This would be good if you were to turn this into a mix task, a full CLI application, or an escript.

### IO Lists:
Using the `IO` module and working with io_lists: 
io_lists are lists of binaries and lists, nested arbitrarily that can just be fed into an `io_device` -- no need to interpolate (And the overhead that contains), it just writes the damn thing to the device. Here's a contrived example: 
```
iex(5)> IO.binwrite([["sup"], "\n", ["Lee", "\n"], ["from", ["\n", ["IO lists"]]]])
sup
Lee
from
IO lists:ok
```
One of the benefits here is that you can stream your output to an IO device (like std_out, or a file) with really really low overhead. It also requires you to deal with a lot more lists and it can be harder to do more complicated formatting.

### Using Docstrings
I haven't been able to get this to work without explicitly adding in newlines, but using a docstring could be more straightforward instead of a bunch of `IO.puts/1` functions:
```
IO.puts("""
============ Results =============\n
Number of records:        #{acc.num_records}\n
Actual number of records: #{actual_number_records}\n
""")
```
I am almost certain there is a way to get this to honor newlines, but I haven't had the time to fiddle with it.